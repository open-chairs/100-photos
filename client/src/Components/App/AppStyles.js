// import styled from 'styled-components';
const {styled} = window;

//minimum w/h => 330 X 280
const PhotoGallery = styled.div`
  padding: 5px;
  justify-content: center;
  max-height: 100%; 
  min-height: 100%; 
  position: relative;
  `;

const Header = styled.h2`
  border-bottom: solid grey 0.2px;
  padding-bottom: 10px;
  font-size: 24px;
  font-weight: 600;
  line-height: 32px;
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
`;

export default { PhotoGallery, Header }