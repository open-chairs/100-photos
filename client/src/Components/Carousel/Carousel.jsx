import React from 'react';
import Styled from './CarouselStyles.js';
import path from 'path';
const flagPath = path.join(__dirname, 'flag.png');


class Carousel extends React.Component {
  constructor(props){
    super(props)
    this.state = {
      photos: this.props.data, 
      currentId: Number(this.props.current),
      currentIndex: 0,
      currentPhoto: {},
      shortDate: ''
    }

    this.leftClickHandler = this.leftClickHandler.bind(this);
    this.rightClickHandler = this.rightClickHandler.bind(this);
  }

  componentDidMount() {
    console.log(this.props.current.split(''));
    let index = Number(this.props.current.split('').slice(5));
    var shortenedDate = this.props.data[index].date.split(' ').splice(0,4).join(' ');

    let newState = {
      currentIndex: index,
      currentPhoto: this.props.data[index],
      shortDate: shortenedDate
    }
      this.setState(newState);
  }

  leftClickHandler(e) {
    if (this.state.currentIndex === 0){
      let newState = {
        currentIndex: this.state.photos.length - 1
      }
      this.setState(newState);      
    } else {
      let newState = {
        currentIndex: this.state.currentIndex - 1
      }
      this.setState(newState);
    }
  }

  rightClickHandler(e) {
    let length = this.state.photos.length - 1
    if (this.state.currentIndex === length){
      let newState = {
        currentIndex: 0
      }
      this.setState(newState);      

    } else {
      let newState = {
        currentIndex: this.state.currentIndex + 1
      }
      this.setState(newState);
    }
  }

  render() 
  {
    return (
      <div>
        <Styled.Main>
                <Styled.LeftButton onClick={this.leftClickHandler}/>
            <Styled.Content>

                  <Styled.PhotoContainer>
                    <Styled.Photo src={this.props.data[this.state.currentIndex].url}></Styled.Photo>
                  </Styled.PhotoContainer>
          
                  <Styled.Description>
                    <Styled.Text>
                        <Styled.Strong>{this.props.data[this.state.currentIndex].description}</Styled.Strong> { this.state.shortDate}
                      <div>
                        {this.props.data[this.state.currentIndex].source}
                      </div>
                      <Styled.Flag src={flagPath}/>
                    </Styled.Text>
                  
                  </Styled.Description>

            </Styled.Content>
                <Styled.RightButton onClick={this.rightClickHandler}/>

        </Styled.Main>
        <Styled.Exit onClick={this.props.toggle}></Styled.Exit>
      </div>
    )
  }
}

export default Carousel;