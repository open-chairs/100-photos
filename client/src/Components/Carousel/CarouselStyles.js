// import styled from 'styled-components';
const {styled} = window;

const Exit = styled.button`
  margin-top: 1px;
  margin-right: 2px;
  position:absolute;
  right: 5px;
  top: 5px;
  background-color: transparent;
  border: none;
  color: white;
  text-decoration: none;
  font-family: icons;
  speak: none;
  font-style: normal;
  font-weight: 400;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "U";
  color: #6f737b;
  font-size: 1.2rem;
  line-height: 2;
  text-align: center;
  top: 16px;
  right: 16px;
  width: 2rem;
  
  :after {
    content: 'X'; 
    font: 16px/100% arial;
  }
`;
  
const Main = styled.div`
  display: grid; 
  grid-template-columns: 25% 50% 25%;
  width: 660px;
  height: auto; 
  min-height: 300px;
  min-height: 300px;
  justify-self: center;
  justify-content: center;
  position: relative;
`;

const Content = styled.div`
  position: relative;
  display: inline; 
  grid-area: auto / 2 / auto / span 1;
  justify-content: center; 
  width: 320px;
  height: 400px
  top: 0;
  left: 0; 
  max-height: 490;
  max-width: 490;
  min-height: 300px;
  min-height: 300px;
`;

const LeftButton = styled.a`
  left: 22%;
  color: #91949a;
  font-size: 1.2rem;
  position: absolute;
  top: 50%;
  margin-left: 15%
  left: 0;
  font-family: icons;
  speak: none;
  font-style: normal;
  font-weight: 100;
  font-variant: normal;
  text-transform: none;
  line-height: 1;

  :after {
    content: '<'; 
  }
`;

const RightButton = styled.a`
  right: 24%;
  color: white; 
  font-size: 1.2rem;
  position: absolute;
  top: 50%;
  margin-right: 60%
  font-family: icons;
  speak: none;
  font-style: normal;
  font-weight: 100;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  
  :after {
    content: '>'; 
  }
`;

const PhotoContainer = styled.div`
display: inline-block; 
position: relative; 
  width: 320px;
  height: 320px;
  max-height: 490;
  max-width: 490;
  min-height: 200px;
  min-height: 200px;
  overflow: hidden; 
`;

const Photo = styled.img`
  justify-content: center;
  overflow: hidden;
  width: 100%;
  height: 100%;
  background-size: cover; 
  position: relative; 
`;

const Description = styled.div`
font-size: 8px;
  color: white; 
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-font-smoothing: antialiased;
  display: block;
`;

const Text = styled.span`
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;

`;

const Strong = styled(Text)`
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 700;
`;

const Flag = styled.a`
  display: inline;
  position: absolute; 
  top: 0
  right: 0;
`;


export default {Exit, Content, Main, Flag, Photo, Text, Strong, PhotoContainer, Description, LeftButton, RightButton}