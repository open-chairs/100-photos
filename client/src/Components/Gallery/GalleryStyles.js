const { styled } = window;

const Gallery = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-content: flex-start; 
  height: 280px; 
  max-width: 626px;

  div.photo {
    margin: .15%;
    box-sizing: border-box; 
    padding: 0; 
    height: 49%;
    width: 22.36%; 
    background-size: cover;
    background-position: center;
    .overlay {
      margin: 0px;
      padding: 0px;
      background-color: rgba(0, 0, 0, .5);
      height: 100%;
      width: 100%;
      position: relative; 
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: center;

      p {
        color: white;
      }
      
    }
    .overlay:hover {
      background-color: rgba(0, 0, 0, 0.65);
    }
  }

  #photo2 {
    height: 98.9%;
    width: 44.72%;
  }

  #photo3, #photo4, #photo5, #photo6, #photo7, #photo8 {
    height: 32.5%;
    width: 14.72%;
  }
`;

const SinglePhoto = styled.img`
  justify-content: center;
  position: relative;
  overflow: hidden;
  width: 100%;
  height: 100%;
  background-size: cover; 
`;

export default {Gallery, SinglePhoto }
