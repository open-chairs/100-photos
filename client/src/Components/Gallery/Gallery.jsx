import React from 'react';
import Styled from './GalleryStyles.js';
import Modal from '../Modal/Modal.jsx';
import Carousel from '../Carousel/Carousel.jsx';

class Gallery extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      currentImgId: 0,
      isOpen: false
    };
    this.toggleModal = this.toggleModal.bind(this);
  }

  toggleModal(e) {
    var id = e.currentTarget.id
    var newState = {
      currentImgId: id,
      isOpen: !this.state.isOpen
    }
    this.setState(newState);
  }

  render() {
    if (this.props.data.length > 9 && this.props.data.length !== 0) {
      return (
        <div>
          <Styled.Gallery>
            {this.props.data.map((item, i) => {
              if (i < 8) {
                return (
                  <div
                    id={`photo${i}`}
                    key={item.id}
                    className="photo"
                    onClick={this.toggleModal}
                    style={{
                      backgroundImage: `url(${item.url})`
                    }}
                  />
                )
              }
              return <div></div>
            })}
            <div
              id="photo8"
              key={this.props.data[8].id}
              className="photo"
              onClick={this.toggleModal}
              style={{
                backgroundImage: `url(${this.props.data[8].url})`
              }}
            >
              <div className="overlay">
                <p>+ {this.props.imgTotal - 9} more</p>
              </div>
            </div>
          </Styled.Gallery>

          <Modal show={this.state.isOpen} onClose={this.toggleModal}>
            <Carousel toggle={this.toggleModal} data={this.props.data} current={this.state.currentImgId} />
          </Modal>

        </div>
      )
    } else if (this.props.data.length <= 9 && this.props.data.length !== 0) {
      return (
        <Styled.Gallery>
          <Styled.SinglePhoto src={this.props.data[0].url}></Styled.SinglePhoto>
        </Styled.Gallery>
      )
    } else if (this.props.data.length === 0) {
      return <div></div>
    }
    else {
      return <div></div>
    }
  }
}




export default Gallery; 