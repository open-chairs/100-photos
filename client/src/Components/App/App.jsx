import React from 'react';
import Styled from './AppStyles.js';
import axios from 'axios';
import Gallery from '../Gallery/Gallery.jsx';
import { relative } from 'path';

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      data: [],
    }
  }

  componentDidMount() {
    let setId, origin; 
    if(window.location.pathname === "/") {
      setId = "/1/"; 
    } else {
      setId = window.location.pathname
    }  
    origin = window.location.origin;
    axios.get(`http://localhost:3002/photos${setId}`)
      .then((response) => {
        let newState = {
          data: response.data
        }
        this.setState(newState);
      })
      .catch(function (error) {
        console.log(error);
      });
  }

  render() {
    var photos = this.state.data; 
    var numPhotos = photos.length;
    var relativeHeader = '';

    if(numPhotos > 100) {
      photos = photos.slice(0, 100);
    }

    if (numPhotos > 1) {
      relativeHeader = numPhotos + ' Photos';
    } else if (numPhotos === 1) {
      relativeHeader = '1 Photo';
    } else if (numPhotos === 0) {
      relativeHeader = '0 Photos'
    }

    return (
      <Styled.PhotoGallery>
        <Styled.Header>{relativeHeader}</Styled.Header>
        <Gallery data={this.state.data} imgTotal={numPhotos} />
      </Styled.PhotoGallery>
    )
  }
}

export default App;