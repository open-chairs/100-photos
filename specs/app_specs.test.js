import React from 'react';
import Enzyme, {shallow} from 'enzyme';

import App from '../client/src/Components/App/App.jsx';
import * as AppStyles from '../client/src/Components/App/AppStyles.js';
import Carousel from '../client/src/Components/Carousel/Carousel.jsx';
import * as CarouselStyles from '../client/src/Components/Carousel/CarouselStyles.js';
import Gallery from '../client/src/Components/Gallery/Gallery';
import * as GalleryStyles from '../client/src/Components/Gallery/GalleryStyles.js'

import Adapter from 'enzyme-adapter-react-16';
import dummyData from '../client/src/sampleData.js';


Enzyme.configure({ adapter: new Adapter() });

describe("App component", () => {
  test("renders", () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.exists()).toBe(true);
  })
  test("mounts", () => {
      const wrapper = shallow(<App/>);
      expect(wrapper.exists()).toBe(true);
  });
  it("works", () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
  test("data in state present after mounting", () => {
    const wrapper = shallow(<App />)
    expect(wrapper.state('data')).toBeTruthy();
    expect(wrapper.state('numOfImages')).toBeGreaterThanOrEqual(0)
  })
});

describe("Gallery component", () => {
  const wrapperMount = shallow(<Gallery/>);
  wrapperMount.setProps = {
    data: dummyData
  }
  test("renders", () => {
      expect(wrapperMount.exists()).toBe(true);
    })
  test("snapshot", () => {
    expect(wrapperMount).toMatchSnapshot()
  });
  test("images are buttons", () => {
    wrapperMount.find('a').simulate('click');
  })
  });


describe("Carousel Component", () => {
  const wrapperMount = shallow(<Carousel/>);
  const wrapperShallow = shallow(<Carousel/>);
  wrapperMount.setProps = {
    data: dummyData
  }
  test("renders", () => {
    expect(wrapperShallow.exists()).toBe(true);
  })
  // test("matches snapshot", () => {
  //   expect(wrapperMount).toMatchSnapshot()
  // })
  test("exit button exists and works", () => {
    expect(
      wrapperMount.find('button').text()
    ).toEqual('X').simulate('click');
  })
  test("include right and left buttons", () => {
    expect(
      wrapperMount.find('button').toEqual('<').simulate('click')
      );
    expect(
      wrapperMount.find('button').toEqual('>').simulate('click')
    );
  })
  test("contains image element", () => {
    expect(
      wrapperMount.find('img').hasClass('CarouselStyles__Photo-ocec9w-6 hBPtId')
    )
  })
});
