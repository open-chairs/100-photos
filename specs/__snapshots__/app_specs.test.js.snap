// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component works 1`] = `
<App>
  <AppStyles__PhotoGallery>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "AppStyles__PhotoGallery-sc-1p402wy-0",
            "isStatic": true,
            "lastClassName": "fIYeEb",
            "rules": Array [
              "padding:5px;justify-content:center;position:static;max-height:100%;max-width:613px;min-height:100%;min-width:352px;position:absolute;",
            ],
          },
          "displayName": "AppStyles__PhotoGallery",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "AppStyles__PhotoGallery-sc-1p402wy-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="AppStyles__PhotoGallery-sc-1p402wy-0 fIYeEb"
      >
        <AppStyles__Header>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "AppStyles__Header-sc-1p402wy-1",
                  "isStatic": true,
                  "lastClassName": "kKBwwF",
                  "rules": Array [
                    "border-bottom:solid grey 0.5px;padding-bottom:10px;font-size:24px;font-weight:600;line-height:32px;font-family:BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;",
                  ],
                },
                "displayName": "AppStyles__Header",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "AppStyles__Header-sc-1p402wy-1",
                "target": "h2",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h2
              className="AppStyles__Header-sc-1p402wy-1 kKBwwF"
            />
          </StyledComponent>
        </AppStyles__Header>
        <Gallery
          data={Array []}
          imgTotal={0}
        >
          <div />
        </Gallery>
      </div>
    </StyledComponent>
  </AppStyles__PhotoGallery>
</App>
`;
